From 74d6dd08d5ab171fabdcfd1596f1185f53768b1c Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Wed, 19 Apr 2023 06:44:32 +0300
Subject: [PATCH] [SW-140835] fwupd: return the correct area address

This patch aligns erom with preboot based on the patch below:
https://gerrit.habana-labs.com/#/c/266784/

module_fwupd_proto_status sets the state->map_offset internally.
and constructing the response FIT, at that address.
the response status message should include one of the options:
0 for no message body,
state->map_offset if message included.

TODO: make module_fwupd_proto_status, more dynamic,
      and check internally if the map_offset is already exist and reuse
      it. Need to make sure that map_offset is zero if no previous use
      has been done during the current transaction

Change-Id: I9c99fa6c06cced6883ad38455a1c70a5398ac8db
---

diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index e996cb5..702770e 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -296,13 +296,11 @@
 		}
 
 		/* if an explicit FIT was attached - use it */
-		if (!offset) {
-			tmp_offset = state->map_offset;
+		if (!tmp_offset) {
 			err = module_fwupd_proto_status(state, NULL);
-
-			/* if prepare fails - just don't include it*/
-			if (err)
-				tmp_offset = 0;
+			/* if prepared ok - just include it */
+			if (!err)
+				tmp_offset = state->map_offset;
 		}
 		/* no break */
 	default:
